engine/gameobject/src/gameobject/comp_anim.cpp

```diff
@@ -514,14 +514,18 @@ namespace dmGameObject
                 duration, delay, animation_stopped, userdata1, userdata2, true);
     }
 
-    static uint32_t GetElementCount(PropertyType type)
+    static uint32_t GetElementCount(PropertyType type, dmhash_t* element_ids)
     {
         switch (type)
         {
         case PROPERTY_TYPE_NUMBER:
             return 1;
         case PROPERTY_TYPE_VECTOR3:
+        {
+            if (element_ids[2] == 0)
+                return 2;
             return 3;
```

Comment:
we don't need new type, we use vector3, but if only 2 element's specified, we use only 2 elements

Reply:
How can we know this? Can't I do a go.animate() of "scale" with the z-component set to 0?

Reply:
`element_id`is an id for a field I set here: https://github.com/defold/defold/blob/1fcad5cf368af8bd6f7234a64e438aa172ec9b64/engine/gameobject/src/gameobject/gameobject.cpp#L3242-L3245

for cases when we need only two components like `xy` I set Z to 0

Reply:
Oh, yes, now I see. 👍